src/main/java/mil/dds/anet/database/PersonDao.java

```diff
@@ -457,4 +459,60 @@ public String clearCache() {
     return AnetConstants.USERCACHE_EMPTY_MESSAGE;
   }
 
+  @InTransaction
+  public int updatePersonHistory(Person p) {
+    final String personUuid = p.getUuid();
+    // Delete old history
+    final int numRows = getDbHandle()
+        .execute("DELETE FROM \"peoplePositions\"  WHERE \"personUuid\" = ?", personUuid);
+    // Add new history
+    for (final PersonPositionHistory history : p.getPreviousPositions()) {
+      updatePeoplePositions(history.getPositionUuid(), personUuid, history.getStartTime(),
+          history.getEndTime());
+    }
+    return numRows;
+  }
+
+  void updatePeoplePositions(final String positionUuid, final String personUuid,
+      final Instant startTime, final Instant endTime) {
+    if (endTime == null) {
+      // we have to make an exception here, as MSSQL has problems inserting a null datetime
+      getDbHandle()
+          .createUpdate("INSERT INTO \"peoplePositions\" "
+              + "(\"positionUuid\", \"personUuid\", \"createdAt\") "
+              + "VALUES (:positionUuid, :personUuid, :createdAt)")
+          .bind("positionUuid", positionUuid).bind("personUuid", personUuid)
+          .bind("createdAt", DaoUtils.asLocalDateTime(startTime)).execute();
+    } else {
+      getDbHandle()
+          .createUpdate("INSERT INTO \"peoplePositions\" "
+              + "(\"positionUuid\", \"personUuid\", \"createdAt\", \"endedAt\") "
+              + "VALUES (:positionUuid, :personUuid, :createdAt, :endedAt)")
+          .bind("positionUuid", positionUuid).bind("personUuid", personUuid)
+          .bind("createdAt", DaoUtils.asLocalDateTime(startTime))
+          .bind("endedAt", DaoUtils.asLocalDateTime(endTime)).execute();
+    }
+  }
+
+  public boolean hasPositionConflict(Person p) {
```

Comment:
Must be `@InTransaction`.
With some small refactoring, method could also be used to replace the very similar `PositionDao::hasPositionConflict`.